unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AvatarShape/AvatarShape.cs

```diff
@@ -0,0 +1,426 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using AvatarSystem;
+using Cysharp.Threading.Tasks;
+using DCL.Components;
+using DCL.Configuration;
+using DCL.Controllers;
+using DCL.Emotes;
+using DCL.Helpers;
+using DCL.Interface;
+using DCL.Models;
+using GPUSkinning;
+using UnityEngine;
+using LOD = AvatarSystem.LOD;
+
+namespace DCL.ECSComponents
+{
+    public interface IAvatarShape
+    {
+        /// <summary>
+        /// Clean up the avatar shape so we can reutilizate it using the pool
+        /// </summary>
+        void Cleanup();
+        
+        /// <summary>
+        /// Apply the model of the avatar, it will reload if necessary
+        /// </summary>
+        /// <param name="scene"></param>
+        /// <param name="entity"></param>
+        /// <param name="newModel"></param>
+        void ApplyModel(IParcelScene scene, IDCLEntity entity, PBAvatarShape newModel);
+
+        /// <summary>
+        /// Get the transform of the avatar shape
+        /// </summary>
+        Transform transform { get; }
+    }
+    
+    public class AvatarShape : MonoBehaviour, IHideAvatarAreaHandler, IPoolableObjectContainer, IAvatarShape, IPoolLifecycleHandler
+    {
+        private const string CURRENT_PLAYER_ID = "CurrentPlayerInfoCardId";
+        private const float MINIMUM_PLAYERNAME_HEIGHT = 2.7f;
+        internal const string IN_HIDE_AREA = "IN_HIDE_AREA";
+
+        // NOTE: Don't use the reference, use the avatarMovementController
+        // Unity can't serialize interfaces, so we assign the reference on the prefab and assign it in the Awake method so we can avoid 1 GetComponent call
+        [SerializeField] private AvatarMovementController avatarMovementControllerReference;
+        internal IAvatarMovementController avatarMovementController;
```

Comment:
Is this really that helpful? The performance improvement is almost not existant (because pools are prewarmed) and we are coupling the systems. This could lead to assembly cycling nightmares.